server/src/main/java/io/crate/execution/dml/upsert/UpdateSourceGen.java

```diff
@@ -88,7 +90,15 @@
         for (String updateColumn : updateColumns) {
             ColumnIdent column = ColumnIdent.fromPath(updateColumn);
             Reference ref = table.getReference(column);
-            this.updateColumns.add(ref == null ? table.getDynamic(column, true) : ref);
+            if (ref != null) {
+                this.updateColumns.add(ref);
+            } else {
+                DynamicReference dynamicReference = table.getDynamic(column, true, true);
+                if (dynamicReference == null) {
+                    throw new ColumnUnknownException(updateColumn,table.ident());
+                }
+                this.updateColumns.add(dynamicReference);
+            }
```

Comment:
Would it be possible to already fail in the analyzer instead of doing this here?

I'm also not sure if this is correct. E.g. `UPDATE t SET some_col['x'] = 10` should probably be possible, even if `x` does not exist?

Reply:
The reason for this change was to eliminate `ColumnUnknownException` from `getDynamic` because I thought returning `null` and the exception are conflicting ideas. It was more for refactoring and did something similar to `InsertAnalyzer`.

> Would it be possible to already fail in the analyzer instead of doing this here?

I now understand that I did not preserve the original behaviour. For the query you mentioned, it would throw the exception at analysis originally, but now it would not. I will revert the changes.

> I'm also not sure if this is correct. E.g. UPDATE t SET some_col['x'] = 10 should probably be possible, even if x does not exist?

This fails on master as well, I will file a bug for this.